File: Paperboy.Api/Dtos/BotDto.cs
using Paperboy.Api.Services;

namespace Paperboy.Api.Dtos;

public class BotDto
{
    public required String Id { get; set; }
    public string? Name { get; set; } = null!;
    public string? Description { get; set; } = null!;
    public string? Status { get; set; } = null!;
    public string? Exchange { get; set; } = null!;
    public string? TradingPair { get; set; } = null!;
    public DateTime? CreatedDate { get; set; } = null;
    public double? StartingBalance { get; set; } = null!;
    public int? TotalTrades { get; set; } = null!;

    public Dictionary<string, OrderDto> Orders { get; set; } = new Dictionary<string, OrderDto>();
}

File: Paperboy.Web/src/scripts/botDto.ts
export class BotDto {
    id: string;
    name?: string;
    description?: string;
    status?: string;
    exchange?: string;
    tradingPair?: string;
    createdDate?: string;
    startingBalance?: number;
    totalTrades?: number;
    orders?: any[];

    constructor(
        id: string,
        name?: string,
        description?: string,
        status?: string,
        exchange?: string,
        tradingPair?: string,
        createdDate?: string,
        startingBalance?: number,
        totalTrades?: number,
        orders?: any[]
    ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.status = status;
        this.exchange = exchange;
        this.tradingPair = tradingPair;
        this.createdDate = createdDate;
        this.startingBalance = startingBalance;
        this.totalTrades = totalTrades;
        this.orders = orders;
    }
}
